Update security policies with bug bounty info (#139)
threshold-network/solidity-contracts: SECURITY.md

# Security Policy

## Bug Bounty Program

Threshold Network has a [Bug Bounty program with Immunefi](https://immunefi.com/bounty/thresholdnetwork/).

> The details for the Bug Bounty are maintained and updated at the [Immunefi dedicated space to Threshold](https://immunefi.com/bounty/thresholdnetwork/). There you can explore the assets in scope of the Bounty and the different rewards by threat level. As a guide, the initial bounty program launched with the following rewards according to the severity of the threats found:

Reviewer: ```suggestion
The details for the Bug Bounty are maintained and updated at the [Immunefi Threshold page](https://immunefi.com/bounty/thresholdnetwork/). There you can explore the assets in scope for the bounty and the different rewards by threat level. As a guide, the initial bounty program launched with the following rewards according to the severity of the threats found:
```

Smart Contracts

- Critical Level: USD $100,000 to USD $500,000
- High Level: USD $10,000 to USD $50,000
- Medium Level: USD $1,000 to USD $5,000
- Low Level: USD $1,000

Websites and Applications

- Critical Level: USD $10,000 to USD $25,000
- High Level: USD $1,000 to USD $10,000
- Medium Level: USD $1,000

> A great place to begin your research is by working on our testnet. Please see our [documentation](https://docs.threshold.network) to get started. We ask that you please respect network machines and their owners. If you find a vulnerability that you suspect has given you access to a machine against the owner's permission, stop what you're doing and create a report using the immunefi dashboard for researchers.

Reviewer: ```suggestion
A great place to begin your research is by working on our testnet. Please see our [documentation](https://docs.threshold.network) to get started. We ask that you please respect network machines and their owners. If you find a vulnerability that you suspect has given you access to a machine against the owner's permission, stop what you're doing and create a report using the Immunefi dashboard for researchers.
```

Rewards are distributed according to the impact of the vulnerability based on the [Immunefi Vulnerability Severity Classification System V2.3](https://immunefi.com/immunefi-vulnerability-severity-classification-system-v2-3/). This is a simplified 4-level scale, with separate scales for websites/apps, smart contracts, and blockchains/DLTs, focusing on the impact of the vulnerability reported.

## Reporting a Vulnerability Not Covered by the Bug Bounty Program

> Please, verify the list of assets in-scope and out-of-scope available as part of the [Threshold Bug Bounty details](https://immunefi.com/bounty/thresholdnetwork/). Additionally, security researchers are encouraged to submit issues outside of the outlined Impacts and Assets in Scope. If you can demonstrate a critical impact on code in production for an asset not in scope, Threshold DAO encourages you to submit your bug report using the “primacy of impact exception” asset in Immunefi.

Reviewer: ```suggestion
Please verify the list of assets in-scope and out-of-scope available as part of the [Threshold Bug Bounty details](https://immunefi.com/bounty/thresholdnetwork/). Additionally, security researchers are encouraged to submit issues outside of the outlined "Impacts" and "Assets in Scope". If you can demonstrate a critical impact on code in production for an asset not in scope, Threshold DAO encourages you to submit your bug report using the “primacy of impact exception” asset in Immunefi.
```

> Threshold DAO will try to make an initial assessment of a bug's relevance, severity, and exploitability, and communicate this back to the reporter. The Threshold DAO will compensate important findings on a case-by-case basis. We value security researchers and we encourage you to contact us to discuss your findings.

Reviewer: Should we mention something about a POC here, or since it is defined with Immunefi we don't need to?

Reviewer: Same conflicting feeling here. Let's leave it out of here for now, so that Immunefi is the canonical source.

Author: I am not opposed to adding something to the topic, but according to immunefi, it needs to be provided or the reports can be closed as not valid. However it did not deter some hackers from submitting low quality proofs. How would you shape it?

# Security Policy

## Bug Bounty Program

Threshold Network has a [Bug Bounty program with Immunefi](https://immunefi.com/bounty/thresholdnetwork/).

The details for the Bug Bounty are maintained and updated at the [Immunefi dedicated space to Threshold](https://immunefi.com/bounty/thresholdnetwork/). There you can explore the assets in scope of the Bounty and the different rewards by threat level. As a guide, the initial bounty program launched with the following rewards according to the severity of the threats found:

Smart Contracts
- Critical Level: USD $100,000 to USD $500,000
- High Level: USD $10,000 to USD $50,000
- Medium Level: USD $1,000 to USD $5,000
- Low Level: USD $1,000

Websites and Applications
- Critical Level: USD $10,000 to USD $25,000
- High Level: USD $1,000 to USD $10,000
- Medium Level: USD $1,000

> **Out of Scope Impacts**
> Please note that the following impacts and attack vectors are excluded from rewards for the Immunefi bug bounty program:

Reviewer: ```suggestion
### Out of Scope Impacts

Please note that the following impacts and attack vectors are excluded from rewards for the Immunefi bug bounty program:
```

# Security Policy

## Bug Bounty Program

> Since April 28th, 2022 Threshold Network has a [Bug Bounty program with Immunefi](https://immunefi.com/bounty/thresholdnetwork/) as approved by Threshold DAO in [TIP-041](https://forum.threshold.network/t/tip-041-establish-a-bug-bounty-program/453) proposal. 

Reviewer 2: ```suggestion
Threshold Network has a [Bug Bounty program with Immunefi](https://immunefi.com/bounty/thresholdnetwork/).
```

Reviewer 2: suggest removing info that's not relevant for security researchers

- Self-XSS
- Captcha bypass using OCR
- CSRF with no security impact (logout CSRF, change language, etc.)
- Missing HTTP Security Headers (such as X-FRAME-OPTIONS) or cookie security flags (such as “httponly”)
- Server-side information disclosure such as IPs, server names, and most stack traces
- Vulnerabilities used to enumerate or confirm the existence of users or tenants
- Vulnerabilities requiring unlikely user actions
- Lack of SSL/TLS best practices
- Attacks involving DOS and/or DDoS
- Attacks that require physical contact to the victims computer and/or wallet
- Attacks requiring privileged access from within the organization
- SPF records for email domains
- Feature requests
- Best practices

Rewards are distributed according to the impact of the vulnerability based on the [Immunefi Vulnerability Severity Classification System V2.2](https://immunefi.com/immunefi-vulnerability-severity-classification-system-v2-2/). This is a simplified 5-level scale, with separate scales for websites/apps, smart contracts, and blockchains/DLTs, focusing on the impact of the vulnerability reported. 

## Reporting a Vulnerability Not Covered by the Bug Bounty Program

> For those assets that are not covered in the Immunefi Bug Bounty program, (please see the updated program [here](https://immunefi.com/bounty/thresholdnetwork/)), if you identify any vulnerabilities within the Threshold Network code and outside our bounty program, please let us know. You can send an email to `security@threshold.network` with relevant information about your findings. We will work with researchers to coordinate vulnerability disclosure between our stakers, partners, and users to ensure the successful mitigation of vulnerabilities.

Reviewer 2: do we prefer people to report out-of-scope issues to security@threshold.network or to just use immunefi anyway since we can escalate/scope in anything that's sufficiently important

Author: I would like to get more input on this issue, that’s a very good question

Reviewer 3: Good point, let's funnel everything through immunefi

> The Threshold team will make a best effort to respond to a new report **within 48 hours**. This response may be a simple acknowledgement that the report was received, or may be an initial assessment from the team. Unless the report is assessed as irrelevant or incorrect, this response will include expected next steps and communication time frames from the Threshold team.

Reviewer 2: ```suggestion
Threshold DAO will make a best effort to respond to a new report **within 48 hours**. This response may be a simple acknowledgement that the report was received, or may be an initial assessment from the team. Unless the report is assessed as irrelevant or incorrect, this response will include expected next steps and communication time frames.
```

> The Threshold team will try to make an initial assessment of a bug's relevance, severity, and exploitability, and communicate this back to the reporter. The Threshold DAO will compensate important findings on a case-by-case basis. We value security researchers and we encourage you to contact us to discuss your findings.

Reviewer 2: ```suggestion
Threshold DAO will try to make an initial assessment of a bug's relevance, severity, and exploitability, and communicate this back to the reporter. The Threshold DAO will compensate important findings on a case-by-case basis. We value security researchers and we encourage you to contact us to discuss your findings.
```
